projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -72,332 +72,299 @@ const PRIORITIZED_INSERTER_BLOCKS = [ ...map( validFields, block => `jetpack/${
-								instanceId={ instanceId }
+						{ 'message' === customThankyou && (
+							<TextareaControl
+								label={ __( 'Message Text', 'jetpack-forms' ) }
+								value={ customThankyouMessage }
+								placeholder={ __( 'Thank you for your submission!', 'jetpack-forms' ) }
+								onChange={ newMessage => setAttributes( { customThankyouMessage: newMessage } ) }
+								__nextHasNoMarginBottom={ true }
 							/>
 						) }
-						{ ! ( isSimpleSite() || isAtomicSite() ) && (
-							<Fragment>
-								{ canUserInstallPlugins && (
-									<PanelBody
-										title={ __( 'CRM Connection', 'jetpack-forms' ) }
-										initialOpen={ false }
-									>
-										<CRMIntegrationSettings
-											jetpackCRM={ jetpackCRM }
-											setAttributes={ setAttributes }
-										/>
-									</PanelBody>
-								) }
-								<PanelBody title={ __( 'Creative Mail', 'jetpack-forms' ) } initialOpen={ false }>
-									<NewsletterIntegrationSettings />
-								</PanelBody>
-							</Fragment>
+
+						{ 'redirect' === customThankyou && (
+							<div>
+								<URLInput
+									label={ __( 'Redirect Address', 'jetpack-forms' ) }
+									value={ customThankyouRedirect }
+									className="jetpack-contact-form__thankyou-redirect-url"
+									onChange={ newURL => setAttributes( { customThankyouRedirect: newURL } ) }
+								/>
+							</div>
 						) }
-					</InspectorControls>
+					</PanelBody>
+					<PanelBody title={ __( 'Email Connection', 'jetpack-forms' ) }>
+						<JetpackEmailConnectionSettings
+							emailAddress={ to }
+							emailSubject={ subject }
+							instanceId={ instanceId }
+							postAuthorEmail={ postAuthorEmail }
+							setAttributes={ setAttributes }
+						/>
+					</PanelBody>
 
-					<div className={ formClassnames } style={ style } ref={ ref }>
-						<InnerBlocks
-							allowedBlocks={ ALLOWED_BLOCKS }
-							prioritizedInserterBlocks={ PRIORITIZED_INSERTER_BLOCKS }
-							templateInsertUpdatesSelection={ false }
+					{ isSalesForceExtensionEnabled && salesforceData?.sendToSalesforce && (
+						<SalesforceLeadFormSettings
+							salesforceData={ salesforceData }
+							setAttributes={ setAttributes }
+							instanceId={ instanceId }
 						/>
-					</div>
-				</>
-			);
-		}
+					) }
+					{ ! ( isSimpleSite() || isAtomicSite() ) && (
+						<>
+							{ canUserInstallPlugins && (
+								<PanelBody title={ __( 'CRM Connection', 'jetpack-forms' ) } initialOpen={ false }>
+									<CRMIntegrationSettings
+										jetpackCRM={ jetpackCRM }
+										setAttributes={ setAttributes }
+									/>
+								</PanelBody>
+							) }
+							<PanelBody title={ __( 'Creative Mail', 'jetpack-forms' ) } initialOpen={ false }>
+								<NewsletterIntegrationSettings />
+							</PanelBody>
+						</>
+					) }
+				</InspectorControls>
 
-		return <div { ...blockProps }>{ elt }</div>;
+				<div className={ formClassnames } style={ style } ref={ innerRef }>
+					<InnerBlocks
```

Comment:
Can probably also reduce these two <div> elements down to one with `useInnerBlocksProps`.

Reply:
That would be good to do. There are just two css rules to change too with `block-editor-inner-blocks`.